File: amazia-core/src/main/java/com/example/operationlog/controller/ListOperationLogController.java
package com.example.operationlog.controller;

import com.example.operationlog.dto.AdminOperationLogItem;
import com.example.operationlog.service.ListOperationLogService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * 操作履歴一覧 API（GET /api/operation-logs）。
 *
 * 想定呼び出し元: Amazia Console（JWT 検証経由）。
 *
 * クエリパラメータ:
 *   - screenName: 部分一致（例: "console.sales_return"）
 *   - apiName   : 部分一致（例: "/api/sales-returns"）
 *   - action    : 完全一致（例: "approve_sales_return"）
 *
 * 命名規約: docs/ai_context/operation_logs_naming.md
 *   action      : (取得系のため operation_logs には記録しない)
 *   screen_name : console.operation_log.list
 *   api_name    : GET /api/operation-logs
 */
@RestController
@RequestMapping("/api")
public class ListOperationLogController {

    private final ListOperationLogService service;

    public ListOperationLogController(ListOperationLogService service) {
        this.service = service;
    }

    @GetMapping("/operation-logs")
    public ResponseEntity<List<AdminOperationLogItem>> list(
            @RequestParam(value = "screenName", required = false) String screenName,
            @RequestParam(value = "apiName",    required = false) String apiName,
            @RequestParam(value = "action",     required = false) String action) {
        return ResponseEntity.ok(service.list(screenName, apiName, action));
    }
}

File: amazia-console/app/OperationLog/Controller/ListOperationLogController.php
<?php

namespace App\OperationLog\Controller;

use App\Http\Controllers\Controller;
use App\OperationLog\Service\ListOperationLogService;
use Illuminate\Http\Request;

/**
 * 操作履歴一覧取得 Controller（GET /api/operation-logs）。
 *
 * Core の GET /api/operation-logs を中継する。
 * 検索クエリ（screenName / apiName / action）はそのまま透過。
 * 認証は routes/api.php 側の auth.jwt ミドルウェアで担保。
 */
class ListOperationLogController extends Controller
{
    private ListOperationLogService $service;

    public function __construct(ListOperationLogService $service)
    {
        $this->service = $service;
    }

    public function __invoke(Request $request)
    {
        $screenName = $request->query('screenName');
        $apiName    = $request->query('apiName');
        $action     = $request->query('action');

        $response = $this->service->list($screenName, $apiName, $action);
        return response()->json($response->json(), $response->status());
    }
}
